### PR TITLE
💄(home) have a cleaner homepage layout

### PR DESCRIPTION
**Pull request by manuhabitela**
Trying a new homepage layout… this makes me create dialogs and forms. It's meant as a starting point to build those low levels components correctly. Kinda wip like your invite poc for now!

https://github.com/user-attachments/assets/72e3598f-2442-48b3-a7fb-307d82905427

### CODE REVIEW

**Comment by lebaudantoine** [comment]
Nice look and feel. The modal/pop-in component and form elements will be extremely useful in the near future.

Few questions:
-  Should we start using the "gauffre" and the AgentConnect official button? Do you feel it's too specific to the French government?
- What do you think about simplifying the UX by placing the "join" input on the same page as the "create a room" button? (I know, Google Driven development lol)

Capture d’écran 2024-07-22 à 16 44 21

**Comment by manuhabitela** [comment]
> Should we start using the "gauffre" and the AgentConnect official button? Do you feel it's too specific to the French government?

Yeah I figured we'd wait until we know more about the overall goal. Let's talk about this next week.

> What do you think about simplifying the UX by placing the "join" input on the same page as the "create a room" button? (I know, Google Driven development lol)

Yeah I figured that too. Visually it would maybe not look as clean since I admit I really don't want inputs without actual labels. The way I did it is, for now, a good excuse to work on dialog and forms as a whole. And another good excuse to teach the user that "another way to join is possible: just use the url", without cluttering the homepage. But for sure these are just ideas.

**Comment by lebaudantoine** [comment]
I'll buy it!

**Comment by manuhabitela** [comment]
@lebaudantoine here is an update, this is ready to merge from my side.


https://github.com/user-attachments/assets/f3ba4d28-fbdf-45c6-a7a9-2be354febfab

File: src/frontend/src/locales/fr/home.json [hunk 1]
```diff
@@ -1,13 +1,13 @@
 {
-  "createMeeting": "Créer une conférence",
+  "createMeeting": "Créer une réunion",
```
Comment (lebaudantoine) [inline src/frontend/src/locales/fr/home.json:2]: luv it this renaming

File: src/frontend/src/features/rooms/utils/isRoomValid.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const roomIdPattern = '[a-z]{3}-[a-z]{4}-[a-z]{3}'
+export const roomRouteRegex = new RegExp(`^[/](?<roomId>${roomIdPattern})$`)
+
+export const isRoomValid = (roomId: string) =>
+  new RegExp(`^${roomIdPattern}$`).test(roomId)
```
Comment (lebaudantoine) [inline src/frontend/src/features/rooms/utils/isRoomValid.ts:5]: cool, nice cleaning

File: src/frontend/src/layout/Header.tsx [hunk 3]
```diff
@@ -1,15 +1,12 @@
...
-import { flex } from '@/styled-system/patterns'
 import { Stack } from '@/styled-system/jsx'
 import { useTranslation } from 'react-i18next'
-import { LanguageSelector } from '@/i18n/LanguageSelector'
-import { A, Badge, Text } from '@/primitives'
-import { authUrl, logoutUrl, useUser } from '@/features/auth'
+import { Text } from '@/primitives'
+import { SettingsButton } from '@/features/settings'
 
 export const Header = () => {
```
Comment (lebaudantoine) [inline src/frontend/src/layout/Header.tsx:9]: > for now! i wanna ditch the header anyway when we rework the conference view

Agree soooo badly, it feels weird

**Comment by lebaudantoine** [comment]
Huge work, thx 🤓
No proper code review, just review the commit messages.  
Features tested locally ✅

**Quick Feedbacks:**

- It seems odd to have the settings icon at the same level as the primary and secondary actions.
- The header with the logo and settings icon is missing from the home page, which feels empty.
- The settings modal mixes account and settings options; I’m not sure if the current icon is the most appropriate, or the dialog has mixed responsibilities, Idk, I wasn't convinced by this organization, I'll get use to it.
- We should avoid offering a logout option during a conference. Wdyt?
- Provide an option to update the display name in the settings (Actually should be doable, let's see how live kit would react)

These are initial thoughts; I can provide more detailed justifications tomorrow.

[My Loom video](https://www.loom.com/share/d9ea43f436bf484eaff8e9131eee3e80?sid=e3c4df62-f89d-4e52-96da-0c4c6ef1b443) overlap with these comments. Nothing blocking, so we can merge your code.

**Review by lebaudantoine — APPROVED** [review]
Good for me

**Comment by manuhabitela** [comment]
Thanks for the feedback. Included some in a new commit :+1:

**Pull request merged**